File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestCharacter.js
print("getCharacter")
if(typeof getCharacter !== 'undefined')
	throw "getCharacter already defined!"
getCharacter = callback("getCharacter")
getCharacterType = typeof getCharacter
print(getCharacterType)
print(getCharacter)
if(getCharacterType !== 'string')
	throw "getCharacter not string!"
callback("setCharacter",getCharacter)

print("getCharacterVector")
if(typeof getCharacterVector !== 'undefined')
	throw "getCharacterVector already defined!"
getCharacterVector = callback("getCharacterVector")
getCharacterVectorType = typeof getCharacterVector[0]
print(getCharacterVectorType)
print(getCharacterVector)
if(getCharacterVectorType !== 'string')
	throw "getCharacterVector not string!"
callback("setCharacterVector",getCharacterVector)

print("getCharacterVectorAsList")
if(typeof getCharacterVectorAsList !== 'undefined')
	throw "getCharacterVectorAsList already defined!"
getCharacterVectorAsList = callback("getCharacterVectorAsList")
getCharacterVectorAsListType = typeof getCharacterVectorAsList[0]
print(getCharacterVectorAsListType)
print(getCharacterVectorAsList)
if(getCharacterVectorAsListType !== 'string')
	throw "getCharacterVectorAsList not string!"
callback("setCharacterVectorAsList",getCharacterVectorAsList)

print("getCharacterMatrix")
if(typeof getCharacterMatrix !== 'undefined')
	throw "getCharacterMatrix already defined!"
getCharacterMatrix = callback("getCharacterMatrix")
getCharacterMatrixType = typeof getCharacterMatrix[0][0]
print(getCharacterMatrixType)
print(getCharacterMatrix)
if(getCharacterMatrixType !== 'string')
	throw "getCharacterMatrix not string!"
callback("setCharacterMatrix",getCharacterMatrix)

print("getCharacterMatrixAsList")
if(typeof getCharacterMatrixAsList !== 'undefined')
	throw "getCharacterMatrixAsList already defined!"
getCharacterMatrixAsList = callback("getCharacterMatrixAsList")
getCharacterMatrixAsListType = typeof getCharacterMatrixAsList[0][0]
print(getCharacterMatrixAsListType)
print(getCharacterMatrixAsList)
if(getCharacterMatrixAsListType !== 'string')
	throw "getCharacterMatrixAsList not string!"
callback("setCharacterMatrixAsList",getCharacterMatrixAsList)
	

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/callback/ParametersAndReturnsTestCharacter.rb
puts("getCharacter")
unless (defined?(getCharacter)).nil? || getCharacter.nil?
	raise "getCharacter already defined!"
end
getCharacter = callback("getCharacter")
getCharacterType = getCharacter.class
puts(getCharacterType)
puts(getCharacter)
if(getCharacterType != Integer)
	raise "getCharacter not Integer"
end
callback("setCharacter", getCharacter)

puts("getCharacterVector")
unless (defined?(getCharacterVector)).nil? || getCharacterVector.nil?
	raise "getCharacterVector already defined!"
end
getCharacterVector = callback("getCharacterVector")
getCharacterVectorType = getCharacterVector[0].class
puts(getCharacterVectorType)
puts(getCharacterVector)
if(getCharacterVectorType != Integer)
	raise "getCharacterVector not Integer"
end
callback("setCharacterVector", getCharacterVector)

puts("getCharacterVectorAsList")
unless (defined?(getCharacterVectorAsList)).nil? || getCharacterVectorAsList.nil?
	raise "getCharacterVectorAsList already defined!"
end
getCharacterVectorAsList = callback("getCharacterVectorAsList")
getCharacterVectorAsListType = getCharacterVectorAsList[0].class
puts(getCharacterVectorAsListType)
puts(getCharacterVectorAsList)
if(getCharacterVectorAsListType != Integer)
	raise "getCharacterVectorAsList not Integer"
end
callback("setCharacterVectorAsList", getCharacterVectorAsList)

puts("getCharacterMatrix")
unless (defined?(getCharacterMatrix)).nil? || getCharacterMatrix.nil?
	raise "getCharacterMatrix already defined!"
end
getCharacterMatrix = callback("getCharacterMatrix")
getCharacterMatrixType = getCharacterMatrix[0][0].class
puts(getCharacterMatrixType)
puts(getCharacterMatrix)
if(getCharacterMatrixType != Integer)
	raise "getCharacterMatrix not Integer"
end
callback("setCharacterMatrix", getCharacterMatrix)

puts("getCharacterMatrixAsList")
unless (defined?(getCharacterMatrixAsList)).nil? || getCharacterMatrixAsList.nil?
	raise "getCharacterMatrixAsList already defined!"
end
getCharacterMatrixAsList = callback("getCharacterMatrixAsList")
getCharacterMatrixAsListType = getCharacterMatrixAsList[0][0].class
puts(getCharacterMatrixAsListType)
puts(getCharacterMatrixAsList)
if(getCharacterMatrixAsListType != Integer)
	raise "getCharacterMatrixAsList not Integer"
end
callback("setCharacterMatrixAsList", getCharacterMatrixAsList)
